tests/test_interpolate.py

```diff
@@ -261,3 +264,38 @@ def test_intermediate_states(num_pairs_to_setup=10):
             xs.append(x0 + 0.01 * np.random.randn(*x0.shape))
         for x in xs:
             np.testing.assert_almost_equal(U_ref(x), U_test(x))
+
+
+def test_duplicate_idxs_period_pairs():
+    """Check that parameter interpolation is able to handle torsion terms with duplicate ((i, j, k, l), period) pairs.
+    E.g. if we only align on idxs and period, this will result in a DuplicateAlignmentKeysError."""
+
+    # CHEMBL3664148
+    mol_a = Chem.AddHs(Chem.MolFromSmiles("CNC(=O)c1cc2cc(Nc3nccc(-c4cn(C)cn4)n3)cc(Cl)c2[nH]1"))
+    AllChem.EmbedMolecule(mol_a)
+
+    # CHEMBL3668838
+    # has a subgroup matching a pattern with duplicate periods:
+    # https://github.com/proteneer/timemachine/blob/c88b42c8aeadaeb7979558f04cae961502fd2917/timemachine/ff/params/smirnoff_2_0_0_ccc.py#L480
+    mol_b = Chem.AddHs(Chem.MolFromSmiles("Cc1cc(Nc2nccc(-c3cn(C)cn3)n2)cc2cc(C(=O)NCc3nccs3)[nH]c12"))
+    AllChem.EmbedMolecule(mol_b)
+
+    core = atom_mapping.get_cores(
```

Comment:
Should the core be fixed (in case we change the atom mapping code later?)

Reply:
Switched to fixed core in https://github.com/proteneer/timemachine/pull/998/commits/f43c120ec90066eca4b415f293bb66d1cd1f38fb